GVRf/Framework/framework/src/main/java/org/gearvrf/GVRShader.java

```diff
@@ -343,7 +348,11 @@ public int bindShader(GVRContext context, GVRShaderData material, String vertexD
 
         synchronized (shaderManager)
         {
-            int nativeShader = addShader(shaderManager, signature, material);
+            int nativeShader = shaderManager.getShader(signature);
+            if (nativeShader > 0)
```

Comment:
Hi @NolaDonato , at line 319  you are checking "if (nativeShader == 0)" after get getShader(...) but at line 352 it has  "if (nativeShader > 0)". 
Just to confirm if it is right because that seems to be some mistake.

Reply:
int nativeShader = shaderManager.getShader(signature);
// getShader will return 0 if the shader has not been added
if (nativeShader == 0)
{
      // addShader will a native shader ID return > 0 if a shader is successfully added.
      nativeShader = addShader(shaderManager, signature, mtl);
}
// If it returns 0, the shader was not created so we don't want to set it on the render data.
if (nativeShader > 0)
{
      rdata.setShader(nativeShader, isMultiview);
}

Reply:
Hi @NolaDonato, the line 319 looks good to me
but I mean about the line 352.

Reply:
You are 100% correct. This looks like a merge problem. It should be == instead of >.